### PR TITLE
Add allow_http documentation to s3 connector

### PR DESCRIPTION
**Pull request by Sevenannn**
## 🗣 Description

* Add description for s3 connector `allow_http` parameter
* Improve MinIO example

### CODE REVIEW

File: spiceaidocs/docs/components/data-connectors/s3.md [hunk 1]
```diff
@@ -60,6 +60,7 @@ Example: `name: cool_dataset`
 - `s3_endpoint`: The S3 endpoint, or equivalent (e.g. MinIO endpoint), for the S3-compatible storage. Defaults to region endpoint. E.g. `s3_endpoint: https://my.minio.server`
 - `s3_region`: Region of the S3 bucket, if region specific. Default value is `us-east-1` E.g. `s3_region: us-east-1`
 - `client_timeout`: Specifies timeout for S3 operations. Default value is `30s` E.g. `client_timeout: 60s`
+- `allow_http`: Allow connection to `s3_endpoint` with `HTTP` protocol. Defaults to `false`
```
Comment (lukekim) [inline spiceaidocs/docs/components/data-connectors/s3.md]:
```suggestion
- `allow_http`: Allow insecure HTTP connections to `s3_endpoint`. Defaults to `false`
```

File: spiceaidocs/docs/components/data-connectors/s3.md [hunk 2]
```diff
@@ -208,8 +209,9 @@ Create a dataset named `cool_dataset` from a Parquet file stored in MinIO.
 - from: s3://s3-bucket-name/path/to/parquet/cool_dataset.parquet
   name: cool_dataset
   params:
-    s3_endpoint: https://my.minio.server
+    s3_endpoint: http://my.minio.server
     s3_region: 'us-east-1' # Best practice for Minio
```
Comment (lukekim) [inline spiceaidocs/docs/components/data-connectors/s3.md]:
```suggestion
    s3_region: 'us-east-1' # Best practice for MinIO
```

**Review by lukekim — DISMISSED** [review]

File: spiceaidocs/docs/components/data-connectors/s3.md [hunk 3]
```diff
@@ -54,22 +54,23 @@ SELECT COUNT(*) FROM cool_dataset;
...
+| `s3_secret`                 | Secret key (e.g. `AWS_SECRET_ACCESS_KEY` for AWS)                                                                                                             |
+| `allow_http`                | Allow insecure HTTP connections to `s3_endpoint`. Defaults to `false`                                                                                         |
 
 For additional CSV parameters, see [CSV Parameters](/reference/file_format.md#csv)
 
 ## Authentication
 
-No authentication is required for public endpoints. For private buckets, set s3_auth to key or iam_role. If using iam_role, the [AWS IAM role](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html) of the running instance is used.
+No authentication is required for public endpoints. For private buckets, set s3_auth to key or iam_role. For Kubernetes Service Accounts with assigned IAM roles, set s3_auth to iam_role. If using iam_role, the [AWS IAM role](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html) of the running instance is used.
```
Comment (eadgbear) [inline spiceaidocs/docs/components/data-connectors/s3.md]:
```suggestion
No authentication is required for public endpoints. For private buckets, set s3_auth to key or iam_role. For Kubernetes Service Accounts with assigned IAM roles, set `s3_auth` to `iam_role`. If using iam_role, the [AWS IAM role](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html) of the running instance is used.
```

Parameter names should always be escaped in backticks

**Review by eadgbear — DISMISSED** [review]
LGTM

**Review by eadgbear — APPROVED** [review]

**Pull request merged**